discovery/proxy_test.go

```diff
@@ -15,42 +17,87 @@ func TestProxyToRendezvous(t *testing.T) {
 	var (
 		topic    = "test"
 		id       = 101
+		limited  = 102
+		limit    = 1
 		reg      = newRegistry()
 		original = &fake{id: 110, registry: reg, started: true}
 		srv      = makeTestRendezvousServer(t, "/ip4/127.0.0.1/tcp/7788")
 		stop     = make(chan struct{})
+		feed     = &event.Feed{}
+		liveness = 100 * time.Millisecond
 		wg       sync.WaitGroup
 	)
 	client, err := rendezvous.NewEphemeral()
 	require.NoError(t, err)
 	reg.Add(topic, id)
+	reg.Add(topic, limited)
 	wg.Add(1)
+	events := make(chan proxyEvent, 10)
+	sub := feed.Subscribe(events)
+	defer sub.Unsubscribe()
 	go func() {
 		defer wg.Done()
-		require.NoError(t, ProxyToRendezvous(original, []ma.Multiaddr{srv.Addr()}, topic, stop))
+		require.NoError(t, ProxyToRendezvous(original, stop, feed, ProxyOptions{
+			Topic:          topic,
+			Servers:        []ma.Multiaddr{srv.Addr()},
+			Limit:          limit,
+			LivenessWindow: liveness,
+		}))
 	}()
-	timer := time.After(3 * time.Second)
-	ticker := time.Tick(100 * time.Millisecond)
+	require.NoError(t, Consistently(func() (bool, error) {
+		records, err := client.Discover(context.TODO(), srv.Addr(), topic, 10)
+		if err != nil && len(records) < limit {
+			return true, nil
+		}
+		if len(records) > limit {
+			return false, fmt.Errorf("more records than expected: %d != %d", len(records), limit)
+		}
+		var proxied Proxied
+		if err := records[0].Load(&proxied); err != nil {
+			return false, err
+		}
+		if proxied[0] != byte(id) {
+			return false, fmt.Errorf("returned %v instead of %v", proxied[0], id)
+		}
+		return true, nil
+	}, time.Second, 100*time.Millisecond))
+	close(stop)
+	wg.Wait()
+	eventSlice := []proxyEvent{}
+	func() {
```

Comment:
Why is there a function that is immediately executed?

Reply:
convenient to return out of it. otherwise, i have to use a flag to track when to stop the for loop